synapse/logging/context.py

```diff
@@ -314,6 +314,15 @@ def __init__(
             # the request param overrides the request from the parent context
             self.request = request
 
+        # if we don't have a `name`, but do have a parent context, use its name.
+        if self.parent_context and not name:
+            name = str(self.parent_context)
+        if not name:
```

Comment:
This will also catch empty strings, which should probably be OK, but not 100% sure. 😄

Reply:
hrm point. I hate it when we do that.